From d578ae65023bb1cc9076cf625e0239c937401791 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Thu, 8 Feb 2024 09:36:11 +0100
Subject: [PATCH] ARM: dts: stm32: remove stm32mp157*-scmi.dtb from compilation

To ease STM32MP157 ST boards rebase, remove stm32mp157*-scmi.dtb from
compilation.
stm32mp157*-scmi.dts will be used (after being renamed into .dtsi) as
include in stm32mp157*.dts.
Due to inclusion in stm32mp157*.dts, there is a special case
regarding stm32mp157c-ev1-scmi.dts(i). Indeed, stm32mp15-scmi.dtsi
include must be also removed to avoid double inclusion (-ev1.dts
includes ed1.dts which includes stm32mp15-scmi.dtsi).
Also remove *-scmi compatible from documentation.

Change-Id: I6651606a2a2462ed8a0140502f2fd14f6c5a22f3
Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
---
 .../devicetree/bindings/arm/stm32/stm32.yaml      |  8 --------
 arch/arm/boot/dts/st/Makefile                     |  4 ----
 ...57a-dk1-scmi.dts => stm32mp157a-dk1-scmi.dtsi} | 13 ++-----------
 ...57c-dk2-scmi.dts => stm32mp157c-dk2-scmi.dtsi} | 13 ++-----------
 ...57c-ed1-scmi.dts => stm32mp157c-ed1-scmi.dtsi} | 13 ++-----------
 ...57c-ev1-scmi.dts => stm32mp157c-ev1-scmi.dtsi} | 15 ++-------------
 6 files changed, 8 insertions(+), 58 deletions(-)
 rename arch/arm/boot/dts/st/{stm32mp157a-dk1-scmi.dts => stm32mp157a-dk1-scmi.dtsi} (78%)
 rename arch/arm/boot/dts/st/{stm32mp157c-dk2-scmi.dts => stm32mp157c-dk2-scmi.dtsi} (80%)
 rename arch/arm/boot/dts/st/{stm32mp157c-ed1-scmi.dts => stm32mp157c-ed1-scmi.dtsi} (80%)
 rename arch/arm/boot/dts/st/{stm32mp157c-ev1-scmi.dts => stm32mp157c-ev1-scmi.dtsi} (77%)

diff --git a/Documentation/devicetree/bindings/arm/stm32/stm32.yaml b/Documentation/devicetree/bindings/arm/stm32/stm32.yaml
index 4bf28e717a56..732041357c14 100644
--- a/Documentation/devicetree/bindings/arm/stm32/stm32.yaml
+++ b/Documentation/devicetree/bindings/arm/stm32/stm32.yaml
@@ -87,26 +87,18 @@ properties:
           - const: st,stm32mp157
 
       - items:
-          - const: st,stm32mp157a-dk1-scmi
           - const: st,stm32mp157a-dk1
           - const: st,stm32mp157
       - items:
-          - const: st,stm32mp157c-dk2-scmi
           - const: st,stm32mp157c-dk2
           - const: st,stm32mp157
       - items:
-          - const: st,stm32mp157c-ed1-scmi
           - const: st,stm32mp157c-ed1
           - const: st,stm32mp157
       - items:
           - const: st,stm32mp157c-ev1
           - const: st,stm32mp157c-ed1
           - const: st,stm32mp157
-      - items:
-          - const: st,stm32mp157c-ev1-scmi
-          - const: st,stm32mp157c-ev1
-          - const: st,stm32mp157c-ed1
-          - const: st,stm32mp157
 
       - description: DH STM32MP1 SoM based Boards
         items:
diff --git a/arch/arm/boot/dts/st/Makefile b/arch/arm/boot/dts/st/Makefile
index 44b264c399ec..5cbd52aa6a14 100644
--- a/arch/arm/boot/dts/st/Makefile
+++ b/arch/arm/boot/dts/st/Makefile
@@ -38,7 +38,6 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp157a-avenger96.dtb \
 	stm32mp157a-dhcor-avenger96.dtb \
 	stm32mp157a-dk1.dtb \
-	stm32mp157a-dk1-scmi.dtb \
 	stm32mp157a-iot-box.dtb \
 	stm32mp157a-microgea-stm32mp1-microdev2.0.dtb \
 	stm32mp157a-microgea-stm32mp1-microdev2.0-of7.dtb \
@@ -49,12 +48,9 @@ dtb-$(CONFIG_ARCH_STM32) += \
 	stm32mp157c-dhcom-pdk2.dtb \
 	stm32mp157c-dhcom-picoitx.dtb \
 	stm32mp157c-dk2.dtb \
-	stm32mp157c-dk2-scmi.dtb \
 	stm32mp157c-ed1.dtb \
-	stm32mp157c-ed1-scmi.dtb \
 	stm32mp157c-emsbc-argon.dtb \
 	stm32mp157c-ev1.dtb \
-	stm32mp157c-ev1-scmi.dtb \
 	stm32mp157c-lxa-mc1.dtb \
 	stm32mp157c-lxa-tac-gen1.dtb \
 	stm32mp157c-lxa-tac-gen2.dtb \
diff --git a/arch/arm/boot/dts/st/stm32mp157a-dk1-scmi.dts b/arch/arm/boot/dts/st/stm32mp157a-dk1-scmi.dtsi
similarity index 78%
rename from arch/arm/boot/dts/st/stm32mp157a-dk1-scmi.dts
rename to arch/arm/boot/dts/st/stm32mp157a-dk1-scmi.dtsi
index c27963898b5e..673b736cc67b 100644
--- a/arch/arm/boot/dts/st/stm32mp157a-dk1-scmi.dts
+++ b/arch/arm/boot/dts/st/stm32mp157a-dk1-scmi.dtsi
@@ -4,15 +4,9 @@
  * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
  */
 
-/dts-v1/;
-
-#include "stm32mp157a-dk1.dts"
 #include "stm32mp15-scmi.dtsi"
 
 / {
-	model = "STMicroelectronics STM32MP157A-DK1 SCMI Discovery Board";
-	compatible = "st,stm32mp157a-dk1-scmi", "st,stm32mp157";
-
 	reserved-memory {
 		optee@de000000 {
 			reg = <0xde000000 0x2000000>;
@@ -55,11 +49,8 @@ &mdma1 {
 	resets = <&scmi_reset RST_SCMI_MDMA>;
 };
 
-&m4_rproc {
-	/delete-property/ st,syscfg-holdboot;
-	resets = <&scmi_reset RST_SCMI_MCU>,
-		 <&scmi_reset RST_SCMI_MCU_HOLD_BOOT>;
-	reset-names =  "mcu_rst", "hold_boot";
+&mlahb {
+	resets = <&scmi_reset RST_SCMI_MCU>;
 };
 
 &rcc {
diff --git a/arch/arm/boot/dts/st/stm32mp157c-dk2-scmi.dts b/arch/arm/boot/dts/st/stm32mp157c-dk2-scmi.dtsi
similarity index 80%
rename from arch/arm/boot/dts/st/stm32mp157c-dk2-scmi.dts
rename to arch/arm/boot/dts/st/stm32mp157c-dk2-scmi.dtsi
index 622618943134..6069c5067114 100644
--- a/arch/arm/boot/dts/st/stm32mp157c-dk2-scmi.dts
+++ b/arch/arm/boot/dts/st/stm32mp157c-dk2-scmi.dtsi
@@ -4,15 +4,9 @@
  * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
  */
 
-/dts-v1/;
-
-#include "stm32mp157c-dk2.dts"
 #include "stm32mp15-scmi.dtsi"
 
 / {
-	model = "STMicroelectronics STM32MP157C-DK2 SCMI Discovery Board";
-	compatible = "st,stm32mp157c-dk2-scmi", "st,stm32mp157";
-
 	reserved-memory {
 		optee@de000000 {
 			reg = <0xde000000 0x2000000>;
@@ -61,11 +55,8 @@ &mdma1 {
 	resets = <&scmi_reset RST_SCMI_MDMA>;
 };
 
-&m4_rproc {
-	/delete-property/ st,syscfg-holdboot;
-	resets = <&scmi_reset RST_SCMI_MCU>,
-		 <&scmi_reset RST_SCMI_MCU_HOLD_BOOT>;
-	reset-names =  "mcu_rst", "hold_boot";
+&mlahb {
+	resets = <&scmi_reset RST_SCMI_MCU>;
 };
 
 &rcc {
diff --git a/arch/arm/boot/dts/st/stm32mp157c-ed1-scmi.dts b/arch/arm/boot/dts/st/stm32mp157c-ed1-scmi.dtsi
similarity index 80%
rename from arch/arm/boot/dts/st/stm32mp157c-ed1-scmi.dts
rename to arch/arm/boot/dts/st/stm32mp157c-ed1-scmi.dtsi
index c7c4d7e89d61..44c817bdf928 100644
--- a/arch/arm/boot/dts/st/stm32mp157c-ed1-scmi.dts
+++ b/arch/arm/boot/dts/st/stm32mp157c-ed1-scmi.dtsi
@@ -4,15 +4,9 @@
  * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
  */
 
-/dts-v1/;
-
-#include "stm32mp157c-ed1.dts"
 #include "stm32mp15-scmi.dtsi"
 
 / {
-	model = "STMicroelectronics STM32MP157C-ED1 SCMI eval daughter";
-	compatible = "st,stm32mp157c-ed1-scmi", "st,stm32mp157";
-
 	reserved-memory {
 		optee@fe000000 {
 			reg = <0xfe000000 0x2000000>;
@@ -60,11 +54,8 @@ &mdma1 {
 	resets = <&scmi_reset RST_SCMI_MDMA>;
 };
 
-&m4_rproc {
-	/delete-property/ st,syscfg-holdboot;
-	resets = <&scmi_reset RST_SCMI_MCU>,
-		 <&scmi_reset RST_SCMI_MCU_HOLD_BOOT>;
-	reset-names =  "mcu_rst", "hold_boot";
+&mlahb {
+	resets = <&scmi_reset RST_SCMI_MCU>;
 };
 
 &rcc {
diff --git a/arch/arm/boot/dts/st/stm32mp157c-ev1-scmi.dts b/arch/arm/boot/dts/st/stm32mp157c-ev1-scmi.dtsi
similarity index 77%
rename from arch/arm/boot/dts/st/stm32mp157c-ev1-scmi.dts
rename to arch/arm/boot/dts/st/stm32mp157c-ev1-scmi.dtsi
index 2ab77e64f1bb..6cd714b9f4fc 100644
--- a/arch/arm/boot/dts/st/stm32mp157c-ev1-scmi.dts
+++ b/arch/arm/boot/dts/st/stm32mp157c-ev1-scmi.dtsi
@@ -4,15 +4,7 @@
  * Author: Alexandre Torgue <alexandre.torgue@foss.st.com> for STMicroelectronics.
  */
 
-/dts-v1/;
-
-#include "stm32mp157c-ev1.dts"
-#include "stm32mp15-scmi.dtsi"
-
 / {
-	model = "STMicroelectronics STM32MP157C-EV1 SCMI eval daughter on eval mother";
-	compatible = "st,stm32mp157c-ev1-scmi", "st,stm32mp157c-ed1", "st,stm32mp157";
-
 	reserved-memory {
 		optee@fe000000 {
 			reg = <0xfe000000 0x2000000>;
@@ -65,11 +57,8 @@ &mdma1 {
 	resets = <&scmi_reset RST_SCMI_MDMA>;
 };
 
-&m4_rproc {
-	/delete-property/ st,syscfg-holdboot;
-	resets = <&scmi_reset RST_SCMI_MCU>,
-		 <&scmi_reset RST_SCMI_MCU_HOLD_BOOT>;
-	reset-names =  "mcu_rst", "hold_boot";
+&mlahb {
+	resets = <&scmi_reset RST_SCMI_MCU>;
 };
 
 &rcc {
-- 
2.39.5

